client/ReviewModule.jsx

```diff
@@ -76,16 +77,19 @@ export default class ReviewModule extends React.Component {
     for (let idx = starNum + 1; idx <= 5; idx += 1) {
       greyArray.push(idx);
     }
-    this.setState({ redVote: redArray, greyVote: greyArray }, () => (
-      console.log(this.state.redVote)));
+    const voteState = { redVote: redArray, greyVote: greyArray };
+    this.handleStateChanges(voteState);
   }
 
-  avgStars() {
+  handleStateChanges(stateChange) {
+    this.setState(stateChange);
+  }
+
+  avgStars(reviewsArray) {
     let sum = 0;
-    const { reviews } = this.state;
-    sum = reviews.reduce((acc, curr) => (acc + curr.stars), 0);
-    const avg = sum / reviews.length;
-    this.setState({ avgStars: avg });
+    sum = reviewsArray.reduce((acc, curr) => (acc + curr.stars), 0);
+    const avg = sum / reviewsArray.length;
```

Comment:
Sorry if its kind of late in the game, but I am wondering if you include any edge case for businesses that have no reviews. In those scenarios, this line will break your code.

Reply:
Good call.